crypto/bls/bls.go

```diff
@@ -18,14 +26,28 @@ func RandPrivateKey() *bls.SecretKey {
 	return &sec
 }
 
+var (
+	errEmptyInput = errors.New("BytesToBLSPublicKey: empty input")
+)
+
 // BytesToBLSPublicKey converts bytes into bls.PublicKey pointer.
 func BytesToBLSPublicKey(bytes []byte) (*bls.PublicKey, error) {
 	if len(bytes) == 0 {
-		return nil, fmt.Errorf("[BytesToBLSPublicKey] bytes is empty")
+		return nil, errEmptyInput
+	}
+	kkey := string(bytes)
+	if k, ok := BLSPubKeyCache.Get(kkey); ok {
+		return k.(*bls.PublicKey), nil
 	}
 	pubKey := &bls.PublicKey{}
 	err := pubKey.Deserialize(bytes)
-	return pubKey, err
+
+	if err != nil {
+		return nil, err
+	}
+
+	BLSPubKeyCache.Add(kkey, pubKey)
```

Comment:
Can we use value instead of pointer for value caching?

Reply:
done